cpp/util_module/algorithm/util.cpp

```diff
@@ -1,20 +1,52 @@
 #include "util.hpp"
 #include "md5.hpp"
 
-void Util::Md5(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+void Util::Md5Procedure(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   mgp::MemoryDispatcherGuard guard{memory};
   const auto arguments = mgp::List(args);
   const auto record_factory = mgp::RecordFactory(result);
   try {
-    const mgp::List arg_list = arguments[0].ValueList();
-    std::string return_string{""};
-    for (auto value : arg_list) {
-      return_string += value.ToString();
+    if (arguments[0].IsString()) {
+      auto record = record_factory.NewRecord();
+      auto string_value = std::string(arguments[0].ValueString());
+      record.Insert(std::string(kArgumentResultMd5).c_str(), md5(string_value));
+    } else if (arguments[0].IsList()) {
+      const mgp::List arg_list = arguments[0].ValueList();
+      std::string return_string{""};
+      for (auto value : arg_list) {
+        return_string += value.ToString();
+      }
+      auto record = record_factory.NewRecord();
+      record.Insert(std::string(kArgumentResultMd5).c_str(), md5(return_string));
+    } else {
```

Comment:
Why are only string and list arguments valid here? The way it’s currently implemented, this module allows hashing lists that contain non-string values, but forbids hashing those exact values outside a list:
```
CALL util_module.md5(1) // error
CALL util_module.md5([1]) // OK
```

Reply:
fixed to accept any parameter